clients/fides-js/src/lib/consent.tsx

```diff
@@ -20,6 +19,16 @@ export const initOverlay = async ({
   debugLog(options.debug, "Initializing Fides consent overlays...");
 
   async function renderFidesOverlay(): Promise<void> {
+    // Check if there are any notices within the experience that do not have a user preference
+    const noticesWithNoUserPreferenceExist: boolean = Boolean(
+      experience?.privacy_notices?.some(
+        (notice) => notice.current_preference == null
+      )
+    );
+    const modalLinkEl = document.getElementById(FIDES_MODAL_LINK);
```

Comment:
to keep the logic together, would it make sense to have the Overlay component do the `document.getElementById`?

Reply:
I have this logic here because I want to be sure we're not rendering the overlay to the DOM in the following case:

- no new notices exist
- link el does not exist in DOM

So that's why I'm trying to keep this check for link element as early as makes sense